src/vm_interp.rs

```diff
@@ -317,16 +317,21 @@ non-issue here, but require explicit handling in FD. This causes a slight
 difference in error checks in CALL_IMM, which we handle in process_result.
 
 [1](https://github.com/firedancer-io/firedancer/blob/93cea434dfe2f728f2ab4746590972644c06b863/src/ballet/sbpf/fd_sbpf_loader.h#L27). */
-fn setup_internal_fn_registry(vm_ctx: &VmContext) -> FunctionRegistry<usize> {
+fn setup_internal_fn_registry(
+    vm_ctx: &VmContext,
+    sbpf_version: SBPFVersion,
+) -> FunctionRegistry<usize> {
     let mut fn_reg = FunctionRegistry::default();
     let max_pc = vm_ctx.rodata.len() / 8;
 
     // register entry point
-    let _ = fn_reg.register_function(
-        ebpf::hash_symbol_name(b"entrypoint"),
-        b"entrypoint",
-        vm_ctx.entry_pc as usize,
-    );
+    let entry_pc = (vm_ctx.entry_pc as usize).min(max_pc.saturating_sub(1));
```

Comment:
What's the reasoning behind bounding the entry PC here? Is this to mimic Agave's behaviour?

Reply:
An invalid entry_pc is discarded by elf parsing. If we leave an invalid entry_pc here, there are situations where solfuzz-agave (the jit) crashes.